src/main/scala/junctions/crossing.scala

```diff
@@ -91,3 +90,46 @@ object ClockToSignal {
     c2s.io.signal_out
   }
 }
+
+object LevelCrossing {
+  class SynchronizerBackend[T <: Data](typ: T, sync: Int, _clock: Clock) extends Module(Some(_clock)) {
+    val io = new Bundle {
+      val in = typ.asInput
+      val out = typ.asOutput
+    }
+
+    io.out := ShiftRegister(io.in, sync)
+  }
+
+  class SynchronizerFrontend[T <: Data](typ: T, _clock: Clock) extends Module(Some(_clock)) {
+    val io = new Bundle {
+      val in = typ.asInput
+      val out = typ.asOutput
+    }
+
+    io.out := RegNext(io.in)
+  }
+
+  def apply[T <: Data](from_clock: Clock, to_clock: Clock, in: T, sync: Int = 3): T = {
```

Comment:
If the user says they want 4 synchronizer stages, give them 4 stages in the target clock domain. The first one in the source clock domain doesn't count.